src/end_entity.rs

```diff
@@ -74,6 +75,37 @@ impl<'a> EndEntityCert<'a> {
         &self.inner
     }
 
+    /// Verifies that the end-entity certificate is valid for use against the
+    /// specified EKU (Extended Key Use).
+    ///
+    /// `supported_sig_algs` is the list of signature algorithms that are
+    /// trusted for use in certificate signatures; the end-entity certificate's
+    /// public key is not validated against this list. `trust_anchors` is the
+    /// list of root CAs to trust. `intermediate_certs` is the sequence of
+    /// intermediate certificates that the server sent in the TLS handshake.
+    /// `time` is the time for which the validation is effective (usually the
+    /// current time).
+    pub fn verify_is_valid_cert_with_eku(
```

Comment:
I think `verify_is_valid_tls_server_cert` can now be expressed with a call to `verify_is_valid_cert_with_eku` passing `ExtendedKeyUsage::RequiredIfPresent(verify_cert::EKU_SERVER_AUTH)` as the `eku`. The body of both functions is the same otherwise.

Perhaps we should allow `verify_is_valid_cert_with_eku` to take a `crls` arg so that `verify_is_valid_tls_client_cert` could also be expressed with `verify_is_valid_cert_with_eku` ? 

In retrospect I'm not sure why I didn't add a `crls` arg to `verify_is_valid_tls_server_cert` when adding CRL support. I don't think we're planning to add server certificate CRL validation to Rustls but the existing CRL validation code in webpki isn't specific to client certificates. We should probably hold off on changing the signature of `verify_is_valid_tls_server_cert` for now since it will be a breaking change but for your new function it seems like a good idea.